### PR TITLE
Refactor search action to extract shared functions

### PR DESCRIPTION
**Pull request by PopDaph**
## Description

Refactoring only, should not change the behavior. 

We will need to create a new mcp internal server to search for global agents. Reason is described here: https://dust4ai.slack.com/archives/C050SM8NSPK/p1749635113038149

This is the first step: moving the search.ts server into a search folder and extracting the `searchFunction` in a util file that will be shared with the new server. 

## Tests

Locally. 
Will also test on front-edge. 

## Risk

Break search action. 
Can be rolled back. 

## Deploy Plan

Deploy front.

### CODE REVIEW

File: front/lib/actions/mcp_internal_actions/servers/search/search.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
+import { INTERNAL_MIME_TYPES } from "@dust-tt/client";
```
Comment (Fraggle) [inline front/lib/actions/mcp_internal_actions/servers/search/search.ts:1]: nit: `search/search.ts` => `search/index.ts`

**Review by Fraggle — APPROVED** [review]

**Review by philipperolet — APPROVED** [review]
:pray:

**Comment by PopDaph** [comment]
tested on front-edge, god is able to do semantic search there seem to be no issue 🙏🏻
